### PR TITLE
Command to amend commits

### PR DESCRIPTION
**Pull request by SiameseCat**
Adding ability to amend commits in monky.

### CODE REVIEW

File: monky.el [hunk 1]
```diff
@@ -2882,6 +2890,19 @@ With a non numeric prefix ARG, show all entries"
       (error "Nothing staged")
     (monky-pop-to-log-edit 'commit)))
 
+(defun monky-commit-amend ()
+  "Amends previous commit.
+Brings up a buffer to allow editing of commit message."
+  (interactive)
+  (if (not (or monky-staged-files (monky-merge-p)))
```
Comment (ananthakumaran) [inline monky.el]: is this necessary? won't this prevent the case where you just want to edit the last commit message

File: monky.el [hunk 2]
```diff
@@ -2882,6 +2890,19 @@ With a non numeric prefix ARG, show all entries"
...
+      (error "Nothing staged")
+	;; get last commit message
+	(with-current-buffer (get-buffer-create monky-log-edit-buffer-name)
+	  (monky-hg-insert
+	   (list "log"
+			 "--template" "{desc}" "-r" "-1")))
```
Comment (ncalexan) [inline monky.el]: I think "-1" should be "." here, for when you haven't checked out tip.  (This is wrong elsewhere in the file, too.)

File: monky.el (hunk 1 above)
Comment (ncalexan) [inline monky.el]: Locally, I dropped this.

**Comment by ananthakumaran** [comment]
Thanks. merged to master.

**Pull request closed without merging**